pysmt/exceptions.py

```diff
@@ -163,12 +163,23 @@ class PysmtUnboundedOptimizationError(PysmtException):
 
 class GoalNotSupportedError(PysmtException):
     """Goal not supported by the solver."""
-    def __init__(self, current_solver, goal):
-        self.current_solver = current_solver
-        self.goal = goal
+    def __init__(self, solver, goal, mode=None):
+        self._solver = solver
+        self._goal = goal
+        self._mode = mode
 
+    @property
     def solver(self):
-        return self.current_solver
+        return self._solver
 
+    @property
     def goal(self):
-        return self.goal
\ No newline at end of file
+        return self._goal
+
+    @property
+    def mode(self):
+        return self._mode
+
+    def __str__(self):
+        mode_str = f" in mode '{self.mode}'" if self.mode else ""
```

Comment:
Pysmt is still compatible with Python 3.5, we disallow f-strings. Please remove them from all your code (check this with `grep -RiI 'f"' pysmt/`)